cargo-dist/templates/installer/installer.sh.j2

```diff
@@ -287,93 +290,84 @@ install() {
-        if [ -n "${CARGO_DIST_FORCE_INSTALL_DIR:-}" ] && [ "${CARGO_DIST_FORCE_INSTALL_DIR}" != "$HOME/.cargo" ]; then
-            _install_dir="$CARGO_DIST_FORCE_INSTALL_DIR/bin"
-            _env_script_path="$CARGO_DIST_FORCE_INSTALL_DIR/env"
-            # For these two, if the new value contains $HOME, make
-            # sure to `sed` the $HOME expression back into the expr
-            _install_dir_expr="$(echo "$CARGO_DIST_FORCE_INSTALL_DIR/bin" | sed "s,$HOME,\$HOME,")"
-            _env_script_path_expr="$(echo "$CARGO_DIST_FORCE_INSTALL_DIR/env" | sed "s,$HOME,\$HOME,")"
-        fi
-    else
-        err "could not find your CARGO_HOME or HOME dir to install binaries to"
-    fi
-{% elif install_path.kind == "HomeSubdir" %}
-    # Install to this subdir of the user's home dir
-    # In this case we want to be late-bound, as $HOME is reliable/nice.
-    if [ -n "${HOME:-}" ]; then
-        _install_dir="$HOME/{{ install_path.subdir }}"
-        _env_script_path="$HOME/{{ install_path.subdir }}/env"
-        _install_dir_expr='$HOME/{{ install_path.subdir }}'
-        _env_script_path_expr='$HOME/{{ install_path.subdir }}/env'
-
-        if [ -n "${CARGO_DIST_FORCE_INSTALL_DIR:-}" ]; then
-            _install_dir="$CARGO_DIST_FORCE_INSTALL_DIR"
-            _env_script_path="$CARGO_DIST_FORCE_INSTALL_DIR/env"
-            # For these two, if the new value contains $HOME, make
-            # sure to `sed` the $HOME expression back into the expr
-            _install_dir_expr="$(echo "$CARGO_DIST_FORCE_INSTALL_DIR" | sed "s,$HOME,\$HOME,")"
-            _env_script_path_expr="$(echo "$CARGO_DIST_FORCE_INSTALL_DIR/env" | sed "s,$HOME,\$HOME,")"
-        fi
-    else
-        err "could not find your HOME dir to install binaries to"
+    {% endfor %}
     fi
-{% elif install_path.kind == "EnvSubdir" %}
-    # Install to this subdir of the user's {{ install_path.env_key }} dir.
-    # In this case we want to be early-bound, as the env-var can't be trusted longterm.
-    if [ -n "{{ "${" }}{{ install_path.env_key }}:-}" ]; then
-        _install_dir="${{ install_path.env_key }}{% if install_path.subdir | length %}/{% endif %}{{ install_path.subdir }}"
-        _env_script_path="${{ install_path.env_key }}{% if install_path.subdir | length %}/{% endif %}{{ install_path.subdir }}/env"
-        _install_dir_expr="$_install_dir"
-        _env_script_path_expr="$_env_script_path"
-
-        # Override if necessary
-        if [ -n "${CARGO_DIST_FORCE_INSTALL_DIR:-}" ]; then
-            _install_dir="$CARGO_DIST_FORCE_INSTALL_DIR"
-            _env_script_path="$CARGO_DIST_FORCE_INSTALL_DIR/env"
-            _install_dir_expr="$_install_dir"
-            _env_script_path_expr="$_env_script_path"
-        fi
-    else
-        err "could not find your {{ install_path.env_key }} dir to install binaries to"
+
+    if [ -z "$_install_dir_expr" ]; then
+        err "could not find a valid path to install to!\n\n$(usage)"
```

Comment:
wait i thought `\n` didn't work in shell

Reply:
Oh god right this is probably completely dependent on which `echo` is being used. I'm sure this isn't portable.